src/Storages/StorageHudi.cpp

```diff
@@ -134,46 +129,36 @@ std::vector<std::string> getKeysFromS3(const StorageS3::S3Configuration & base_c
 }
 
 
-StorageS3Configuration getAdjustedS3Configuration(
-    StorageS3::S3Configuration & base_configuration,
-    const StorageS3Configuration & configuration,
```

Comment:
I mostly merged these `StorageS3::S3Configuration` and `StorageS3Configuration` into one, because this looks very bad and inconvenient.

Reply:
I have unified StorageHudi/DeltaLake/Iceberg into one base class, and also TableFunctionHudi/DeltaLake/Iceberg in https://github.com/ClickHouse/ClickHouse/pull/45384, could you take a look at first?

Reply:
@ucasfl, this PR is actually a reason of me reading your PR today. While reading your PR It struck me how inconvenient it is that there is this code I mentioned.